### PR TITLE
Add `rb-connection_pool`

### PR DESCRIPTION
**Pull request by svengiegerich**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/rb-connection_pool/LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+Copyright (c) 2011 Mike Perham
```
Comment (janjagusch) [inline recipes/rb-connection_pool/LICENSE.txt:1]: @svengiegerich, you should be able to get rid of the license file by following the same steps we discussed here: https://github.com/conda-forge/staged-recipes/pull/22205

File: recipes/rb-connection_pool/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE.txt
+  summary: Generic connection pool for Ruby
+  doc_url: https://www.rubydoc.info/gems/{{ name }}
+  dev_url: https://github.com/mperham/{{ name }}
+
+extra:
+  recipe-maintainers:
+    - svengiegerich
```
Comment (janjagusch) [inline recipes/rb-connection_pool/meta.yaml:40]: happy to co-maintain. :)

```suggestion
    - svengiegerich
    - janjagusch
```

**Comment by janjagusch** [comment]
> @janjagusch how can I add two packages within the same PR (as one depends on the other)? I've tried in [e52addb](https://github.com/conda-forge/staged-recipes/commit/e52addb71afaab1c42f7756e1a5d66527f7c72cb) but failed with the msg that the dependency cannot be found... 😶‍🌫️ (I'd like to do that so that we don't need another iteration cycle for approvals...)

Just check `rb-net-http-persistent/meta.yaml` into this branch, and then CI should build both packages.

File: recipes/rb-net-http-persistent/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+requirements:
+  host:
+    - ruby <2.6, >=2.3
+  run:
+    - ruby <2.6, >=2.3
+
+test:
+  commands:
+    - ruby -r {{ name }} -e 'exit 0'
+    - rb-connection_pool >=2.2, <3
```
Comment (janjagusch) [inline recipes/rb-net-http-persistent/meta.yaml:28]: You probably added this by accident, right?

```suggestion
```

File: recipes/rb-net-http-persistent/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+requirements:
+  host:
+    - ruby <2.6, >=2.3
+  run:
+    - ruby <2.6, >=2.3
+
+test:
+  commands:
+    - ruby -r {{ name }} -e 'exit 0'
```
Comment (janjagusch) [inline recipes/rb-net-http-persistent/meta.yaml:27]: I think the import for this package is different, see [here](https://github.com/drbrain/net-http-persistent#synopsis-).

```suggestion
    - ruby -r 'net/http/persistent' -e 'exit 0'
```

File: recipes/rb-net-http-persistent/LICENSE.txt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+(The MIT License)
```
Comment (janjagusch) [inline recipes/rb-net-http-persistent/LICENSE.txt:1]: @svengiegerich, you should be able to get rid of the license file. :)

**Review by janjagusch — APPROVED** [review]

File: recipes/rb-connection_pool/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    - gem install -N -l -V --norc --ignore-dependencies {{ name }}-{{ version }}.gem
+    - gem unpack {{ name }}-{{ version }}.gem
+    - cp {{ name }}-*/LICENSE .
+  skip: true  # [win]
+
+requirements:
+  host:
+    - ruby <2.6
+  run:
+    - ruby <2.6
```
Comment (xhochy) [inline recipes/rb-connection_pool/meta.yaml:24]: For now, let's pin ruby more explicit.
```suggestion
    - ruby >=2.5,<2.6
  run:
    - ruby >=2.5,<2.6
```

**Comment by xhochy** [comment]
I would prefer if this could be ported to a newer Ruby version but it seems that our general Ruby stack is way behind and thus this is probably a reasonable compromise for now.

**Pull request merged**